src/test/java/teammates/test/cases/logic/EmailGeneratorTest.java

```diff
@@ -0,0 +1,255 @@
+                      + "{*}" + course.getId() + "{*}" + course.getName()
+                      + "{*}" + session.getFeedbackSessionName()
+                      + "{*}" + submitUrl + "{*}" + submitUrl
+                      + "{*}You may ignore this email if you have already submitted feedback.";
+        
+        verifyEmail(emails.get(0), student1.email, subject, textInEmail);
+        
+        verifyEmail(emails.get(1), student3.email, subject, "The following feedback session is closing soon");
+        verifyEmail(emails.get(1), student3.email, subject,
+                    "You may ignore this email if you have already submitted feedback.");
+        
+        ______TS("feedback session published alerts");
+        
+        emails = new EmailGenerator().generateFeedbackSessionPublishedEmails(session);
+        assertEquals(10, emails.size());
+        
+        subject = String.format(EmailType.FEEDBACK_PUBLISHED.getSubject(),
+                                course.getName(), session.getFeedbackSessionName());
+        
+        String reportUrl = Config.getAppUrl(Const.ActionURIs.STUDENT_FEEDBACK_RESULTS_PAGE)
+                                 .withCourseId(course.getId())
+                                 .withSessionName(session.getFeedbackSessionName())
+                                 .withRegistrationKey(StringHelper.encrypt(student1.key))
+                                 .withStudentEmail(student1.email)
+                                 .toAbsoluteString();
+        
+        textInEmail = "Hello " + student1.name + "{*}The feedback responses for the following "
+                      + "feedback session is now open for viewing{*}" + course.getId()
+                      + "{*}" + course.getName() + "{*}" + session.getFeedbackSessionName()
+                      + "{*}" + reportUrl + "{*}" + reportUrl;
+        
+        verifyEmail(emails.get(0), student1.email, subject, textInEmail);
+        
+        verifyEmail(emails.get(1), student2.email, subject,
+                    "The feedback responses for the following feedback session is now open for viewing");
+    }
+    
+    @Test
+    public void testGenerateStudentCourseJoinEmail() throws IOException, MessagingException {
+        
+        CourseAttributes course = new CourseAttributes("course-id", "Course Name");
+        
+        StudentAttributes student = new StudentAttributes();
+        student.name = "Student Name";
+        student.key = "skxxxxxxxxxks";
+        student.email = "student@email.tmt";
+        
+        MimeMessage email = new EmailGenerator().generateStudentCourseJoinEmail(course, student);
+        String subject = String.format(EmailType.STUDENT_COURSE_JOIN.getSubject(), course.getName(), course.getId());
+        
+        // check email body
+        String joinUrl = Config.getAppUrl(student.getRegistrationUrl()).toAbsoluteString();
+        String textInEmail = "Hello " + student.name + "{*}course <i>" + course.getName()
```

Comment:
Very minor point, and you can ignore this if you want, but can we close the `<i>` tag that's opened here. It looks very wrong that we only check for the opening `<i>`